compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -569,6 +569,12 @@ impl<'a> Parser<'a> {
         lhs_span: Span,
         expr_kind: fn(P<Expr>, P<Ty>) -> ExprKind,
     ) -> PResult<'a, P<Expr>> {
+        let lhs_span = lhs
```

Comment:
Please deduplicate this code with all other similar sites (e.g. https://github.com/rust-lang/rust/blob/6a54bfc44b5a1b47506fcdf4f074671328d963dd/compiler/rustc_parse/src/parser/expr.rs#L251) by putting it in a function and calling it at all those sites

Reply:
Thanks! Good spot and requested change. The overall diff is small enough that I elected to just squash everything into a single commit instead of adding a separate one with the updates, hope that's alright

Reply:
Noticed the same issue could occur on range expressions, so added a fix for that one as well (though we haven't had any issue reports for range expressions like we have for casts)

Reply:
That's great, thanks for looking into other sites, too!